stories/components/TimePickerField/TimePickerField.stories.tsx

```diff
@@ -0,0 +1,127 @@
+        },
+    },
+};
+
+export const TwelveHourInteractionTest = Default.bind({});
+
+TwelveHourInteractionTest.args = {
+    testID: 'timepickerfield',
+    is24Hour: false,
+};
+
+TwelveHourInteractionTest.parameters = {
+    chromatic: { disableSnapshot: true },
+};
+
+TwelveHourInteractionTest.play = async ({ canvasElement }) => {
+    const canvas = await within(canvasElement);
+
+    await waitFor(() => canvas);
+
+    const timeInput = await canvas.getByTestId('timepickerfield-flat');
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '1');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('01:00am')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '00:30');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('12:30am')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '1230pm');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('12:30pm')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    // for hour the first digit cannot be greater than 1 and the second digit cannot be greater than 2
+    await userEvent.type(timeInput, '3213130');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('11:30am')).toBeInTheDocument());
+};
+
+export const TwentyFourHourInteractionTest = Default.bind({});
+
+TwentyFourHourInteractionTest.args = {
+    testID: 'timepickerfield',
+    is24Hour: true,
+};
+
+TwentyFourHourInteractionTest.parameters = {
+    chromatic: { disableSnapshot: true },
+};
+
+TwentyFourHourInteractionTest.play = async ({ canvasElement }) => {
+    const canvas = await within(canvasElement);
+
+    await waitFor(() => canvas);
+
+    const timeInput = await canvas.getByTestId('timepickerfield-flat');
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '1');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('01:00')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '00:30');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('00:30')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    await userEvent.type(timeInput, '1230pm');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('12:30')).toBeInTheDocument());
+
+    await userEvent.clear(timeInput);
+    // for hour the first digit cannot be greater than 2 and the second digit cannot be greater than 3
+    await userEvent.type(timeInput, '3213130');
+    await userEvent.click(canvasElement);
+
+    await waitFor(() => expect(canvas.getByDisplayValue('21:31')).toBeInTheDocument());
```

Comment:
can we verify that if you paste this value in the field it still gives us 21:31?